### DISCUSSION TITLE
Automatically Taking Pictures with GoPro Based on a Boolean Variable

### DISCUSSION THREAD

**Discussion by Ahmad-197** [gopro/labs#566] [Q&A]
I am using a GoPro affixed to a robot. Whenever the robot travels a predetermined distance the Raspberry Pi would receive a signal. I want my GoPro to take a photo whenever that signal is received. How can I achieve this? I know GoPro can take photos based on timer but can it also take photos based on a boolean variable? Any help or suggestion would be appreciated.

**Comment by dnewman-gpsw** [comment]
In Labs if you can which USB power on and off, that can be used as a control trigger. Of course with the Open GoPro API you can fully control the camera via USB or BLE.

**Comment by Ahmad-197** [comment]
Hi, I am new to it. Can you elaborate it a bit further? What steps should I follow?

**Reply by dnewman-gpsw** [reply]
I can't elaborate too much on Open GoPro, that is for developers for integrating GoPro camera in there products. Robotics seems well suited for Open GoPro.  https://gopro.github.io/OpenGoPro/

If your Raspberry Pi can switch a relay to add / remove power or a USB line, then a Labs script can used to trigger any type of capture including photos.

Using the new visual scripting tool. from: https://gopro.github.io/labs/build/
![image](https://github.com/gopro/labs/assets/10856084/0cc92e8c-1d0b-44e4-be58-493b34ec36b2)

This above is a Photo trigger on the rising edge of add USB power. It generates this script:
![USB Power Photo Trigger-9_28_2023, 09_45_05](https://github.com/gopro/labs/assets/10856084/fa70258f-ed5c-4ad6-8c52-0249a2fde9aa)

Third option: Get the Remote, and have the robot press the record button. ;)
